src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -1101,6 +1096,46 @@ private static ExecutorService getExecutorService() {
         return new WorkerThreadPoolExecutor(threadFactory);
     }
 
+    private static void setRegionForClient(@NonNull final Object client, @NonNull final String region) {
```

Comment:
You might be able to simplify all of this to something like:
``` java
private <S, T> void trySetField(S source, String field, Consumer<T> f, T value) {
        try {
            f.accept(value);
        } catch (UnsupportedOperationException uoe) {
            LOG.debug("Write not supported on " + source.getClass().getSimpleName() + " for " + field);
        }
    }
```
called from the constructor like
``` java
trySetField(kinesisClient, "region", kinesisClient::setRegion, region);
trySetField(dynamoDBClient, "region", dynamoDBClient::setRegion, region);
```